web-services/query/src/main/java/datawave/webservice/query/remote/RemoteQueryServiceImpl.java

```diff
@@ -0,0 +1,370 @@
+package datawave.webservice.query.remote;
+
+import com.codahale.metrics.Counter;
+import com.fasterxml.jackson.databind.ObjectReader;
+import datawave.security.authorization.DatawavePrincipal;
+import datawave.webservice.common.remote.RemoteHttpService;
+import datawave.webservice.common.remote.RemoteQueryService;
+import datawave.webservice.result.BaseQueryResponse;
+import datawave.webservice.result.DefaultEventQueryResponse;
+import datawave.webservice.result.GenericResponse;
+import datawave.webservice.result.VoidResponse;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.PostConstruct;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+public class RemoteQueryServiceImpl extends RemoteHttpService implements RemoteQueryService {
+    private static final Logger log = LoggerFactory.getLogger(RemoteQueryServiceImpl.class);
+    
+    private static final String CREATE = "%s/create";
+    
+    private static final String NEXT = "%s/next";
+    
+    private static final String CLOSE = "%s/close";
+    
+    private static final String PLAN = "%s/plan";
+    
+    private ObjectReader voidResponseReader;
+    
+    private ObjectReader genericResponseReader;
+    
+    private ObjectReader baseQueryResponseReader;
+    
+    private ObjectReader eventQueryResponseReader;
+    
+    private boolean useSrvDNS = false;
+    
+    private List<String> srvDnsServers = Collections.singletonList("127.0.0.1");
+    
+    private int srvDnsPort = 8600;
+    
+    private String queryServiceScheme = "https";
+    
+    private String queryServiceHost = "localhost";
+    
+    private int queryServicePort = 8443;
+    
+    private String queryServiceURI = "/query/v1/";
+    
+    private int maxConnections = 100;
+    
+    private int retryCount = 5;
+    
+    private int unavailableRetryCount = 15;
+    
+    private int unavailableRetryDelay = 2000;
```

Comment:
Consider a separate configuration object to hold these fields?

Reply:
perhaps.  considering.

Reply:
done